### PR TITLE
React: Hook up add participants submit button

### LINKED ISSUE #222: Add participants: hook up submit button

### PR DESCRIPTION
**Pull request by frewmack**
Closes #222.
The `condition` field was also changed to be required on frontend with default value `GermLine`, since it's set as not nullable.

### CODE REVIEW

File: react/src/pages/addParticipants/AddParticipants.tsx [hunk 1]
```diff
@@ -27,16 +32,43 @@ const useStyles = makeStyles(theme => ({
...
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify(data),
+        });
+
+        if (response.ok) {
+            const responseData: Array<any> = await response.json();
+            enqueueSnackbar(`${responseData.length} participants successfully added.`, {
```
Comment (kevinlul) [inline react/src/pages/addParticipants/AddParticipants.tsx:55]: Should be datasets that are added

File: react/src/pages/utils/functions.tsx [hunk 2]
```diff
@@ -251,3 +252,20 @@ export function createFieldObj(
...
+
+    var arr = [];
+    for (let i = 0; i < amount; i++) {
+        arr.push({
+            family_codename: "",
+            participant_codename: "",
+            participant_type: "",
+            tissue_sample_type: "",
+            dataset_type: "",
+            condition: "GermLine",
```
Comment (kevinlul) [inline react/src/pages/utils/functions.tsx:267]: I'll wait until standup but I'm not sure if this is the best default choice, if there even should be one.

**Review by kevinlul — CHANGES_REQUESTED** [review]
Can we verify that the required fields are filled out before trying to submit?

File: react/src/pages/addParticipants/AddParticipants.tsx [hunk 3]
```diff
@@ -64,24 +64,58 @@ export default function AddParticipants() {
...
+                    if (problemRows.get(i)) problemRows.set(i, problemRows.get(i)!.concat(field));
+                    else problemRows.set(i, [field]);
+                }
+            }
+        }
+
+        if (problemRows.size > 0) {
+            let errorMessage = "Cannot submit. Required fields missing for rows:";
+            problemRows.forEach((fields, key) => {
+                const fieldStr = fields.toString().replaceAll(/\[\]/g, "");
```
Comment (kevinlul) [inline react/src/pages/addParticipants/AddParticipants.tsx:88]: Does this show something different in your browser? I expect `Array.toString()` to behave the same as `Array.join()` so we shouldn't have any square brackets in it to begin with.

Comment (frewmack) [reply react/src/pages/addParticipants/AddParticipants.tsx:88]: Actually I didn't consider using `.join()` for some reason. I'll do that instead.

**Review by kevinlul — APPROVED** [review]
LGTM

**Pull request merged**